pytorch_lightning/accelerators/gpu.py

```diff
@@ -47,9 +47,10 @@ def on_train_start(self) -> None:
         with torch.cuda.device(self.root_device):
             torch.cuda.empty_cache()
 
-    def on_train_end(self) -> None:
+    def teardown(self) -> None:
+        self.lightning_module.cpu()
```

Comment:
@SeanNaren for sharded training, do we need to move both self.model and self.lightning_module? Could this lead to errors like what we saw for eval/train with SDP earlier?